controllers/datadogagent/feature/admissioncontroller/feature.go

```diff
@@ -0,0 +1,161 @@
+	"github.com/DataDog/datadog-operator/apis/datadoghq/v2alpha1"
+	apiutils "github.com/DataDog/datadog-operator/apis/utils"
+	componentdca "github.com/DataDog/datadog-operator/controllers/datadogagent/component/clusteragent"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/feature"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func init() {
+	err := feature.Register(feature.AdmissionControllerIDType, buildAdmissionControllerFeature)
+	if err != nil {
+		panic(err)
+	}
+}
+
+type admissionControllerFeature struct {
+	mutateUnlabelled       bool
+	serviceName            string
+	agentCommunicationMode string
+	localServiceName       string
+
+	serviceAccountName string
+	owner              metav1.Object
+}
+
+func buildAdmissionControllerFeature(options *feature.Options) feature.Feature {
+	return &admissionControllerFeature{}
+}
+
+// ID returns the ID of the Feature
+func (f *admissionControllerFeature) ID() feature.IDType {
+	return feature.AdmissionControllerIDType
+}
+
+func (f *admissionControllerFeature) Configure(dda *v2alpha1.DatadogAgent) (reqComp feature.RequiredComponents) {
+	f.owner = dda
+	f.serviceAccountName = v2alpha1.GetClusterAgentServiceAccount(dda)
+
+	ac := dda.Spec.Features.AdmissionController
+	if ac != nil && apiutils.BoolValue(ac.Enabled) {
+		f.mutateUnlabelled = apiutils.BoolValue(ac.MutateUnlabelled)
+		if ac.ServiceName != nil && *ac.ServiceName != "" {
+			f.serviceName = *ac.ServiceName
+		}
+		if ac.AgentCommunicationMode != nil && *ac.AgentCommunicationMode != "" {
+			f.agentCommunicationMode = *ac.AgentCommunicationMode
+		}
+		f.localServiceName = v2alpha1.GetLocalAgentServiceName(dda)
+		reqComp = feature.RequiredComponents{
+			ClusterAgent: feature.RequiredComponent{IsRequired: apiutils.NewBoolPointer(true)},
+		}
+	}
+	return reqComp
+}
+
+func (f *admissionControllerFeature) ConfigureV1(dda *v1alpha1.DatadogAgent) (reqComp feature.RequiredComponents) {
+	// f.owner = dda
+	// f.serviceAccountName = v1alpha1.GetClusterAgentServiceAccount(dda)
+
+	// if dda.Spec.ClusterAgent.Config != nil && dda.Spec.ClusterAgent.Config.AdmissionController != nil && apiutils.BoolValue(dda.Spec.ClusterAgent.Config.AdmissionController.Enabled) {
+	// 	ac := dda.Spec.ClusterAgent.Config.AdmissionController
+	// 	f.mutateUnlabelled = apiutils.BoolValue(ac.MutateUnlabelled)
+	// 	f.serviceName = *ac.ServiceName
+	// 	if ac.AgentCommunicationMode != nil && *ac.AgentCommunicationMode != "" {
+	// 		f.agentCommunicationMode = *ac.AgentCommunicationMode
+	// 	}
+	// 	f.localServiceName = v1alpha1.GetLocalAgentServiceName(dda)
+	// 	reqComp = feature.RequiredComponents{
+	// 		ClusterAgent: feature.RequiredComponent{IsRequired: apiutils.NewBoolPointer(true)},
+	// 	}
+	// }
+	// return reqComp
+
+	// do not apply this feature on v1alpha1
+	// it breaks the unittests in `controller_test.go` because the `store` modifies
```

Comment:
Is this something that we can fix in this PR or does it require too many changes?